### PR TITLE
Feature/e UI 4639 jurisdiction config

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4615

### Change description ###
Getting jurisdictions from config

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: api/supportedJurisdictions/index.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { NextFunction, Response, Router } from 'express';
+import { getConfigValue } from '../configuration';
+import { WA_SUPPORTED_JURISDICTIONS } from '../configuration/references';
+import { EnhancedRequest } from '../lib/models';
+
+export async function getWASupportedJurisdictions(req: EnhancedRequest, res: Response, next: NextFunction): Promise<any> {
+  try {
+    if (!req.session.supportedJurisdictions) {
+      const jurisdictions = getConfigValue(WA_SUPPORTED_JURISDICTIONS);
+      req.session.supportedJurisdictions = jurisdictions;
```
Comment (udaydenduluri33) [inline api/supportedJurisdictions/index.ts:10]: No advantage storing it in session.

File: api/workAllocation2/index.ts [hunk 2]
```diff
@@ -186,7 +187,9 @@ export async function retrieveAllCaseWorkers(req: EnhancedRequest, res: Response
     return req.session.caseworkers;
   }
   const roleApiPath: string = prepareRoleApiUrl(baseRoleAssignmentUrl);
-  const payload = prepareRoleApiRequest();
+  const jurisdictions = req.session && req.session.supportedJurisdictions
```
Comment (udaydenduluri33) [inline api/workAllocation2/index.ts:190]: Please remove code that storing supportedJurisdictions in session

File: api/workAllocation2/utils.spec.ts [hunk 3]
```diff
@@ -176,6 +177,27 @@ describe('workAllocation.utils', () => {
 
   });
 
+  describe('prepareRoleApiRequest', () => {
+
+    it('should correctly prepare a payload with jursidictions and location parameters', () => {
```
Comment (kuda1992) [inline api/workAllocation2/utils.spec.ts:182]: spelling mistake its jurisdictions

File: src/work-allocation-2/containers/all-work-case/all-work-case.component.ts [hunk 4]
```diff
@@ -56,7 +57,7 @@ export class AllWorkCaseComponent extends WorkCaseListWrapperComponent implement
   public backUrl: string = 'work/all-work/cases';
 
   public ngOnInit(): void {
-    this.setUpLocations();
+    this.setUpLocationsAndJurisdicttions();
```
Comment (kuda1992) [inline src/work-allocation-2/containers/all-work-case/all-work-case.component.ts:60]: another spelling mistake.

**Review by kuda1992 — APPROVED** [review]
Minor spelling mistakes, but looks good to me

File: api/workAllocation2/utils.spec.ts (hunk 3 above)
Comment (connorpgpmcelroy) [reply api/workAllocation2/utils.spec.ts:182]: thanks for noticing, will fix

File: src/work-allocation-2/containers/all-work-case/all-work-case.component.ts (hunk 4 above)
Comment (connorpgpmcelroy) [reply src/work-allocation-2/containers/all-work-case/all-work-case.component.ts:60]: will fix

**Review by udaydenduluri33 — APPROVED** [review]
@connorpgpmcelroy  Thanks for making the changes

**Pull request merged**